### ISSUE TITLE
[rush] provide github action

### ISSUE THREAD

**Issue by muuvmuuv** [microsoft/rushstack#2057]
## Please prefix the issue title with the project name i.e. [rush], [api-extractor] etc. ##

**Is this a feature or a bug?**

- [x] Feature
- [ ] Bug

**Please describe the actual behavior.**

I am giving rush a try in my github profile repo and currently setting it up to my needs. To benefit from rush build in github actions I would like to use a rush action which sets itself up and I just need to `run` the rush command I need. This would add great benefits for software hosted on github.

I am open for alternatives :)

**Comment by octogonz** [comment]
>  use a rush action which sets itself up

The typical alternative is [install-run-rush.js](https://rushjs.io/pages/maintainer/enabling_ci_builds/) which will automatically install the correct version of the Rush engine for the branch.

That said, if someone wants to create a GitHub action for Rush, that would be even better!

**Comment by muuvmuuv** [comment]
Yes, that is the way I do it ATM: https://github.com/muuvmuuv/muuvmuuv/blob/master/.github/workflows/age.yml

**Comment by renoirb** [comment]
What would be useful is to see how RushStack phblishes and builds itself with GitHub actions, alongside the use of ´dependabot ´.
Maybe I'm overthinking something that can be simple.
Has there been plans for this?
Or knowledge of technical limitations? Or known caveats (documented?)

**Comment by reggi** [comment]
@renoirb I too am looking for ways to use tools like `dependabot` and `semantic-release` as well as run my testing suite pre-publish with rush, but only running tests for package that have been updated, and dealing with parallelization.

**Comment by renoirb** [comment]
On thet thread about *dependabot*, I've opened a request and I've been suggested to point out to dependabot/dependabot-core#2270 so I'm making a link between this thread and the issue I've been refered to

**Comment by kierans** [comment]
> Yes, that is the way I do it ATM: https://github.com/muuvmuuv/muuvmuuv/blob/master/.github/workflows/age.yml

@muuvmuuv Sadly your example is broken as shown in your GitHub actions logs https://github.com/muuvmuuv/muuvmuuv/runs/2812006949?check_suite_focus=true

```
/home/runner/work/_temp/24eba50c-5f38-416a-be02-0709ee052fbe.sh: line 1: rush: command not found
Error: Process completed with exit code 127.
```

This is the problem I'm having, trying to find the `rush` CLI tool post install to then use rush in a workflow.

**Comment by shawnmclean** [comment]
Has anyone successfully used github actions to run a rush publish command?

Mine seems to not be using the generated `/common/temp/publish-home/.npmrc` file.

**Comment by kierans** [comment]
I have a step in https://github.com/RedCrewOS/api-sdk-creator-js/blob/main/.github/workflows/ci.yml

```
- name: Publish
  if: ${{ github.ref == 'refs/heads/main' }}
  run: rush publish -p -b main --include-all --set-access-level=public
  env:
    NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```

**Comment by arunvishnun** [comment]
Hey @kierans - Thanks for sharing the above GithubAction. 
For some reason, I get below error - 
```
npm notice 
npm ERR! code ENEEDAUTH
npm ERR! need auth This command requires you to be logged in to https://registry.npmjs.org/
npm ERR! need auth You need to authorize this machine using `npm adduser`
```

I know this is a bit old post, but how was the NPM_TOKEN set up ? Was it under "Repository secrets" in settings ? Or any clu how this can be fixed ?

**Comment by kamontat** [comment]
> Hey @kierans - Thanks for sharing the above GithubAction. For some reason, I get below error - I have NPM_TOKEN setup under "Repository secrets" in settings.
> 
> ```
> npm notice 
> npm ERR! code ENEEDAUTH
> npm ERR! need auth This command requires you to be logged in to https://registry.npmjs.org/
> npm ERR! need auth You need to authorize this machine using `npm adduser`
> ```
> 
> I know this is a bit old post, but how was the NPM_TOKEN set up ? Was it under "Repository secrets" in settings ? Or any clu how this can be fixed ?

First please ensure that you have `common/config/rush/.npmrc-publish` file.
Then ensure below code is not commented.

```
//registry.npmjs.org/:_authToken=${NPM_TOKEN}
```

Note: Ensure that environment name is matches with your environment, because as I recall the default env is `NPM_AUTH_TOKEN`
